pypeit/core/coadd.py

```diff
@@ -766,19 +799,36 @@ def sn_weights(waves, fluxes, ivars, masks, sn_smooth_npix, const_weights=False,
             weights[:, iexp] = sn_conv
     else:
         weights = np.zeros_like(flux_stack)
+        refscale = (sn_val[:, ref_spec] > 0) / (sn_val[:, ref_spec] + (sn_val[:, ref_spec] == 0))
```

Comment:
refscale is only needed if relative_weights == True and so it should be computed in the if relative_weights if block above, not here.